.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,32 @@
+name: Upload Python package
+
+on:
+  release:
+    types: [published]
+
+
+jobs:
+  ci:
+    uses: ./.github/workflows/ci-workflow.yml
+
+  build:
+    runs-on: ubuntu-latest
+    needs: ci
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
```

Comment:
Ah yes, you're probably right. The list of supported Python versions shown in pypi is some setup.py metadata rather than something detected from the package.